airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/state/MemoryManager.kt

```diff
@@ -17,19 +17,19 @@ import kotlin.concurrent.withLock
  * TODO: Some degree of logging/monitoring around how accurate we're actually being?
  */
 @Singleton
-class MemoryManager {
-    private val availableMemoryBytes: Long = Runtime.getRuntime().maxMemory()
+class MemoryManager(availableMemoryProvider: AvailableMemoryProvider) {
+    private val availableMemoryBytes: Long = availableMemoryProvider.availableMemoryBytes
```

Comment:
really should be totalMemoryBytes. Once they're reserved, they're not really available anymore

Reply:
agreed done